backend/store/etcd/event_store.go

```diff
@@ -161,7 +161,7 @@ func (s *Store) GetEventByEntityCheck(ctx context.Context, entityName, checkName
 
 // UpdateEvent updates an event.
 func (s *Store) UpdateEvent(ctx context.Context, event *types.Event) error {
-	if event.Check == nil {
+	if event == nil || event.Check == nil {
 		return errors.New("event has no check")
```

Comment:
Is that the correct error for the `event == nil` case? I think it should be something along the lines of "cannot update nil event" or something.

Reply:
The == nil case is really just a guard against a panic, not sure it deserves its own error case.